File: урок2/openspace.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question('', (n) => {
    rl.question('', (input) => {
        const arr = input.split(' ').map(Number);
        
        const prefSums = new Array(Number(n) + 1).fill(0);
        for (let i = 1; i <= n; i++) {
            prefSums[i] = prefSums[i - 1] + arr[i - 1];
        }

        const suffSums = new Array(Number(n) + 1).fill(0);
        for (let i = n - 1; i >= 0; i--) {
            suffSums[i] = suffSums[i + 1] + arr[i];
        }

        let ans = 0;
        for (let i = 1; i < n; i++) {
            ans += arr[i] * i;
        }

        let nowSum = ans;
        for (let nowPos = 1; nowPos < n; nowPos++) {
            nowSum += prefSums[nowPos] - suffSums[nowPos];
            ans = Math.min(ans, nowSum);
        }

        console.log(ans);
        rl.close();
    });
});

File: урок2/openspace.py
n = int(input())
arr = list(map(int, input().split()))
pref_sums =[0] * (n + 1)
for i in range(1, n + 1):
    pref_sums[i] = pref_sums[i - 1] + arr[i - 1]
suff_sums = [0] * (n + 1)
for i in range(n - 1, -1, -1):
    suff_sums[i] = suff_sums[i + 1] + arr[i]
ans = 0
for i in range(1, n):
    ans += arr[i] * i
now_sum = ans
for now_pos in range(1, n):
    now_sum += pref_sums[now_pos] - suff_sums[now_pos]
    ans = min(ans, now_sum)
print(ans)